run_cpp_backend.py

```diff
@@ -17,11 +17,14 @@
     DN = '.' # so that we always have an executable path like ./usercode.exe
 USER_PROGRAM = sys.argv[1] # string containing the program to be run
 LANG = sys.argv[2] # 'c' for C or 'cpp' for C++
+USER_INPUT = ""
+
+prettydump = ('--prettydump' in sys.argv)
 
-prettydump = False
 if len(sys.argv) > 3:
-    if sys.argv[3] == '--prettydump':
-        prettydump = True
+    thrid_argument_is_input = (sys.argv[3] != '--prettydump')
+    if thrid_argument_is_input:
+        USER_INPUT = sys.argv[3]
```

Comment:
I'm not sure if I understand this well, but are you sending the entire STDIN in a command line argument? This seems inefficient and error-prone. For example, how do you make sure that the contents of the input are not interpreted as a fourth argument when you send it from the command line? IMHO a better approach is saving the input in a file, and sending the file name as a command-line argument